frontend/src/features/properties/map/research/update/models.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  fromApiOrganization,
+  fromApiPerson,
+  IContactSearchResult,
+  toOrganization,
+  toPerson,
+} from 'interfaces';
+import { Api_ResearchFile, Api_ResearchFilePurpose } from 'models/api/ResearchFile';
+
+export class ResearchFilePurposeFormModel {
+  public id?: string;
+  public researchPurposeTypeCode?: string;
+  public researchPurposeTypeDescription?: string;
+
+  public static fromApi(base: Api_ResearchFilePurpose): ResearchFilePurposeFormModel {
+    var newModel = new ResearchFilePurposeFormModel();
+    newModel.id = base.id;
+    newModel.researchPurposeTypeCode = base.researchPurposeTypeCode?.id;
+    newModel.researchPurposeTypeDescription = base.researchPurposeTypeCode?.description;
+    return newModel;
+  }
+
+  public toApi(): Api_ResearchFilePurpose {
+    return {
+      id: this.id,
+      researchPurposeTypeCode: {
+        id: this.researchPurposeTypeCode,
+        description: this.researchPurposeTypeDescription,
+      },
+    };
+  }
+}
+
+export class UpdateResearchFormModel {
+  public id?: number;
+  public name?: string;
+  public roadName?: string;
+  public roadAlias?: string;
+  public rfileNumber?: string;
+  public statusTypeCode?: string;
+  public statusTypeDescription?: string;
+  //public researchProperties?: Api_ResearchFileProperty[];
+  public requestDate?: string;
+  public requestDescription?: string;
+  public requestSourceDescription?: string;
+  public researchResult?: string;
+  public researchCompletionDate?: string;
+  public isExpropriation?: string;
+  public expropriationNotes?: string;
+  public requestSourceTypeCode?: string;
+  public requestSourceTypeDescription?: string;
+  public requestor?: IContactSearchResult;
+  public researchFilePurposes?: ResearchFilePurposeFormModel[];
+  public rowVersion?: number;
+
+  public static fromApi(base: Api_ResearchFile): UpdateResearchFormModel {
+    var model = new UpdateResearchFormModel();
+    model.id = base.id;
+    model.name = base.name;
+    model.roadName = base.roadName;
+    model.roadAlias = base.roadAlias;
+    model.rfileNumber = base.rfileNumber;
+    model.statusTypeCode = base.statusTypeCode?.id;
+    model.statusTypeDescription = base.statusTypeCode?.description;
+    //model.researchProperties = base.researchProperties;
+    model.requestDate = base.requestDate;
+    model.requestDescription = base.requestDescription;
+    model.requestSourceDescription = base.requestSourceDescription;
+    model.researchResult = base.researchResult;
+    model.researchCompletionDate = base.researchCompletionDate;
+    model.isExpropriation = base.isExpropriation ? 'true' : 'false';
```

Comment:
surprised by this, are we displaying these as strings?

Reply:
This was necessary since the current select component requires a string for its value.